src/python/pants/backend/python/rules/pex_from_target_closure.py

```diff
@@ -45,7 +47,17 @@ class CreatePexFromTargetClosure:
     chrooted_sources: Optional[ChrootedPythonSources] = None
     if request.include_source_files:
         chrooted_sources = await Get[ChrootedPythonSources](HydratedTargets(all_targets))
-
+    input_digests = []
+    if request.additional_input_files:
+        input_digests.append(request.additional_input_files)
+    if request.include_source_files:
+        chrooted_sources = await Get[ChrootedPythonSources](HydratedTargets(all_targets))
+        input_digests.append(chrooted_sources.snapshot.directory_digest)
+    merged_input_digest: Optional[Digest] = None
+    if input_digests:
+        merged_input_digest = await Get[Digest](
+            DirectoriesToMerge(directories=tuple(input_digests))
+        )
```

Comment:
No need for the `if` statement. If `input_digests` is empty, then `DirectoriesToMerge` will return `EMPTY_DIGEST`.

```python
test = await Get[Digest](DirectoriesToMerge(()))
print(test)

> Digest(fingerprint='e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', serialized_bytes_length=0)
```